pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/sink/PulsarSink.java

```diff
@@ -400,7 +415,16 @@ public void close() throws Exception {
         ConsumerConfig consumerConfig = new ConsumerConfig();
         consumerConfig.setSchemaProperties(pulsarSinkConfig.getSchemaProperties());
         if (!StringUtils.isEmpty(pulsarSinkConfig.getSchemaType())) {
-            consumerConfig.setSchemaType(pulsarSinkConfig.getSchemaType());
+            if (GenericRecord.class.isAssignableFrom(typeArg)) {
+                consumerConfig.setSchemaType(SchemaType.AUTO_CONSUME.toString());
+                SchemaType configuredSchemaType = SchemaType.valueOf(pulsarSinkConfig.getSchemaType());
+                if (SchemaType.AUTO_CONSUME != configuredSchemaType) {
```

Comment:
Need to add `consumerConfig.setSchemaType(pulsarSinkConfig.getSchemaType());` in this branch?

Reply:
No. The schema type is already overwritten in line 419. This is just to log an info message to indicate that the schema type has been overwritten to `AUTO_CONSUME`.